UI/MapWnd.cpp

```diff
@@ -283,6 +284,198 @@ namespace {
+            LayoutRow("MAP_FLEET_PART_COUNT",       label_ul, label_lr, value_ul, value_lr, GG::Pt(GG::X0, GG::Y0));
+
+            Resize(GG::Pt(value_lr.x + (m_margin * 3), value_lr.y + (m_margin * 3)));
+        }
+
+        /** Constructs and attaches new label and value controls for the given label. */
+        void NewLabelValue(const std::string& label, bool title = false) {
+            if (m_labels.find(label) != m_labels.end())
+                return;
+
+            GG::Y height{ClientUI::Pts()};
+            // center format for title label
+            m_labels.emplace(label, std::make_pair(
+                std::shared_ptr<CUILabel>{
+                    new CUILabel{UserString(label),
+                        title ? GG::FORMAT_CENTER : GG::FORMAT_RIGHT,
+                        GG::NO_WND_FLAGS, GG::X0, GG::Y0,
+                        m_col_widths.at(0) - (m_margin * 2), height
+                    }
+                },
+                std::shared_ptr<CUILabel>{
+                    new CUILabel{"0", GG::FORMAT_RIGHT,
+                        GG::NO_WND_FLAGS, GG::X0, GG::Y0,
+                        m_col_widths.at(1) - (m_margin * 2), height
+                    }
+                }
+            ));
+
+            if (title) {  // utilize bold font for title label and value
+                m_labels.at(label).first.get()->SetFont(ClientUI::GetBoldFont());
+                m_labels.at(label).second.get()->SetFont(ClientUI::GetBoldFont());
+            }
+
+            AttachChild(m_labels.at(label).first.get());
+            AttachChild(m_labels.at(label).second.get());
+        }
+
+        /** Updates the text displayed for the value of each label */
+        void UpdateLabels() {
+            UpdateValues();
+            for (const std::unordered_map<std::string, int>::value_type& value : m_values) {
+                std::unordered_map<std::string, LabelValueType>::iterator label_it = m_labels.find(value.first);
+                if (label_it == m_labels.end())
+                    continue;
+                label_it->second.second->ChangeTemplatedText(boost::lexical_cast<std::string>(value.second), 0);
+            }
+        }
+
+    protected:
+        /** Updates the value for display with each label */
+        void UpdateValues() {
+            m_values.clear();
+            m_values["MAP_FLEET_SHIP_COUNT"] = 0;
+            if (m_empire_id == ALL_EMPIRES)
+                return;
+
+            const std::set<int>& destroyed_objects = GetUniverse().EmpireKnownDestroyedObjectIDs(m_empire_id);
+            for (std::shared_ptr<const Ship> ship : Objects().FindObjects<Ship>()) {
+                if (!ship->OwnedBy(m_empire_id) || destroyed_objects.find(ship->ID()) != destroyed_objects.end())
+                    continue;
+                m_values["MAP_FLEET_SHIP_COUNT"]++;
+
+                if (ship->IsArmed())
+                    m_values["MAP_FLEET_ARMED_COUNT"]++;
```

Comment:
all these string literals appearing in several places should be put into constants defined in one place above the relevant code.